ibis/expr/api.py

```diff
@@ -227,23 +225,6 @@ def param(type: dt.DataType) -> ir.Scalar:
     return ops.ScalarParameter(type).to_expr()
 
 
-# TODO(kszucs): should be deprecated
```

Comment:
Shall we have a `4.2` release where we can deprecate this function and mark for removal in `5.0`?

Reply:
We can also deprecate in 5 and remove in 6.

Reply:
Which one would you prefer?

Reply:
Let's deprecate in 5 and remove in 6.

Reply:
Updated, the implementation was incorrect actually.